fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/RecordQuery.java

```diff
@@ -170,11 +172,39 @@ public Builder toBuilder() {
         return new Builder(this);
     }
 
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof RecordQuery)) {
+            return false;
+        }
+        final RecordQuery that = (RecordQuery)o;
+        return isSortReverse() == that.isSortReverse() &&
+               removeDuplicates == that.removeDuplicates &&
+               getRecordTypes().equals(that.getRecordTypes()) &&
+               Objects.equals(getAllowedIndexes(), that.getAllowedIndexes()) &&
+               Objects.equals(queryabilityFilter, that.queryabilityFilter) &&
+               Objects.equals(getFilter(), that.getFilter()) &&
+               Objects.equals(getSort(), that.getSort()) &&
+               Objects.equals(getRequiredResults(), that.getRequiredResults());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getRecordTypes(), getAllowedIndexes(), queryabilityFilter, getFilter(), getSort(), isSortReverse(), removeDuplicates, getRequiredResults());
```

Comment:
Should this use a `hashCodeSupplier` similar to the `BoundRecordQuery`?  I guess it will depend on how often this is used in the process of comparing against cache entries?

Reply:
I think we are ok here without a lazy supplier as the one in `BoundRecordQuery` does the heavy-lifting for this one. The reason the other one needs it more is because that one also does some graph walking which I would like avoid if I can.